src/builtins/impls/urlquery.rs

```diff
@@ -16,28 +16,70 @@
 
 use std::collections::HashMap;
 
-use anyhow::{bail, Result};
+use serde_json::Value;
+use url::form_urlencoded::{byte_serialize, parse};
+
+const QUERY_CONCAT_CHAR: &str = "&";
 
 /// Decodes a URL-encoded input string.
-#[tracing::instrument(name = "urlquery.decode", err)]
-pub fn decode(x: String) -> Result<String> {
-    bail!("not implemented");
+#[tracing::instrument(name = "urlquery.decode")]
+pub fn decode(x: String) -> String {
+    parse(x.as_bytes())
+        .map(|(key, val)| [key, val].concat())
+        .collect()
 }
 
 /// Decodes the given URL query string into an object.
-#[tracing::instrument(name = "urlquery.decode_object", err)]
-pub fn decode_object(x: String) -> Result<HashMap<String, Vec<String>>> {
-    bail!("not implemented");
+#[tracing::instrument(name = "urlquery.decode_object")]
+pub fn decode_object(x: String) -> HashMap<String, Vec<String>> {
```

Comment:
Here you should also definitely use `form_urlencoded::parse`

Something like

```rust
let parsed = parse(x.as_bytes()).into_owned();
let mut decoded_object: HashMap<String, Vec<String>> = HashMap::new();
for (k, v) in parsed {
  decoded_object
    .entry(k)
    .or_default()
    .push(v);
}
```

Reply:
you right, looks better now.